otx/core/action_dataset_adapter.py

```diff
@@ -0,0 +1,169 @@
+        However, Since overall labels are [0, 1, 2, 4], 'video2/frame_000.jpg' should has the label index as 3.
+        
+        """
+        outputs = {
+            "category_items": [],
+            "label_groups" : [],
+            "label_entities": [],
+        }
+
+        category_list = [] # to check the duplicate case
+        for cvat_data in datumaro_dataset[Subset.TRAINING]:
+            # Making overall categories
+            categories = cvat_data.categories().get(DatumaroAnnotationType.label, None)
+            
+            if categories not in category_list:
+                outputs["category_items"].extend(categories.items)
+                outputs["label_groups"].extend(categories.label_groups)
+            
+            category_list.append(categories)
+
+            # Reindexing the each label index of multiple video datasets
+            for cvat_data_item in cvat_data:
+                for ann in cvat_data_item.annotations:
+                    ann_name = categories.items[ann.label].name
+                    ann.label = [i for i, category_item in enumerate(outputs["category_items"]) if category_item.name == ann_name][0]
+
+        # Generate label_entity list according to overall categories 
+        outputs["label_entities"] = [LabelEntity(name=class_name.name, domain=self.domain,
+                            is_empty=False, id=ID(i)) for i, class_name in enumerate(outputs["category_items"])]
+        return outputs
+
+
+class ActionClassificationDatasetAdapter(ActionBaseDatasetAdapter, BaseDatasetAdapter):
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> DatasetEntity:
+        label_information = self._prepare_label_information(datumaro_dataset)
+        label_entities = label_information["label_entities"]
+
+        label_schema = self._generate_default_label_schema(label_entities)
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for datumaro_items in subset_data:
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
+                    shapes = []
+                    for ann in datumaro_item.annotations:
+                        # Action Classification
+                        if isinstance(ann, DatumaroLabel):
+                            shapes.append(
+                                Annotation(
+                                    Rectangle.generate_full_box(),
+                                    labels = [
+                                        ScoredLabel(
+                                            label=label_entities[ann.label]                                        )
+                                    ]
+                                )
+                            )
+                    # Unlabeled dataset
+                    if len(shapes) == 0:
+                        annotation_scene = NullAnnotationSceneEntity()
```

Comment:
this could be good to have discussion for filtering out the null annotated media.

Reply:
Make sense, we need to discuss more for empty(null) labels